bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/SortedLedgerStorage.java

```diff
@@ -78,6 +79,7 @@ public void initialize(ServerConfiguration conf,
                 .setNameFormat("SortedLedgerStorage-%d")
                 .setPriority((Thread.NORM_PRIORITY + Thread.MAX_PRIORITY) / 2).build());
         this.stateManager = stateManager;
+        this.isTransactionalCompactionEnabled = conf.getUseTransactionalCompaction();
```

Comment:
Separate change, so should not be in this patch.

Reply:
this is part of this change itself. Provided more comments for why it is needed - https://github.com/apache/bookkeeper/pull/1281/files#diff-c5b1fb6226c382c12ed9f2dd9e6c8ed2R226

Reply:
I don't see what the point of this flag even is. The external behaviour of SortedLedgerStorage should not change with this patch, so why the flag?

Reply:
@ivankelly 

> The external behaviour of SortedLedgerStorage should not change with this patch

it is not an external behavior of SortedLedgerStorage. It is the logic of SortedLedgerStorage. SortedLedgerStorage needs to realize the compaction mechanism to do checkpointing. That's related to the discussion below around checkpointing.

Reply:
If you have to supply a flag, then there's something external to be observed.

Let me rephrase. As I understand it, we do not want the behaviour of SortedLedgerStorage flushing to change because of the change in this PR. So why are we adding a new configuration flag?